astroid/interpreter/objectmodel.py

```diff
@@ -584,7 +598,10 @@ def attr___name__(self):
     @property
     def attr___doc__(self):
         return node_classes.Const(
-            value=self._instance.parent.doc, parent=self._instance
+            value=self._instance.parent.doc_node.value
+            if self._instance.parent.doc_node
+            else None,
```

Comment:
```suggestion
            value=getattr(self._instance.parent.doc_node, "value", None),
```
Not the best solution but a bit less ugly.